module/zfs/zfs_ioctl.c

```diff
@@ -3082,8 +3082,9 @@ zfs_fill_zplprops_impl(objset_t *os, uint64_t zplver,
 
 	ASSERT(zplprops != NULL);
 
+	/* parent dataset must be a filesystem */
 	if (os != NULL && os->os_phys->os_type != DMU_OST_ZFS)
-		return (SET_ERROR(EINVAL));
+		return (SET_ERROR(ENOTDIR));
```

Comment:
Not sure if this is right. It looks like this will be interpretted by userspace as a "wrong parent" error, but this error has nothing to do with the parent.

Reply:
@tcaputi unless i am misunderstanding the code, `os` is the "parent objset pointer", so if `os->os_phys->os_type != DMU_OST_ZFS` the parent is technically "wrong".

Reply:
You are correct. sorry for the noise